tools/autograd/templates/Functions.cpp

```diff
@@ -314,6 +314,13 @@ Tensor split_backward(const std::vector<torch::autograd::Variable> &grads, int64
   return ret;
 }
 
+Tensor adaptive_max_pool_double_backward(const Tensor & grad, const Tensor & self, const Tensor & indices) {
+  auto size = std::vector<int64_t>(indices.sizes().slice(0, 2));
```

Comment:
is this right?  AdaptiveMaxPool2d accepts 3D and 4D inputs, AdaptiveMaxPool3d accepts 4D and 5D inputs.  So I don't see how this can be a constant.  We probably don't have tests though...